### ISSUE TITLE
Test new terminal reconnection

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#131453]
Refs: #116113

- [ ] macOS
- [ ] linux
- [ ] windows

Complexity: 3

Authors: @Tyriar, @meganrogge 

---

Terminal reconnection has changed, instead of recording a bunch of resize/data events we now run xterm.js on the pty host and leverage the serialize addon to send over a minimal amount of state which improves start up time and allows fine tuning of how much the user wants to restore upon reconnection.

Test the following:

- Review settings `terminal.integrated.persistentSessionExperimentalSerializer` and `terminal.integrated.persistentSessionScrollback`
- Reload the window and make sure the buffer is restored correctly, try just a regular shell as well as various apps like tmux/vim
- Disable `terminal.integrated.persistentSessionExperimentalSerializer`, restart VS Code, fill terminal buffer, reload and make sure ~1000 rows of scrollback are restored

**Comment by weinand** [comment]
The ´terminal.integrated.persistentSessionScrollback` setting makes sense to me as a user visible configuration.
But `terminal.integrated.persistentSessionExperimentalSerializer` is just for giving users a chance to go back in case of errors, right? Do we really have to surface this setting (and provide translations etc.)? Shouldn't the test pass find errors in the new serializer?

**Comment by Tyriar** [comment]
@weinand yes that setting is there just in case something goes wrong, the plan is to remove it next month. It's there to mitigate risk and think it's justified.
